File: js/problem55.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    if (nums.length < 2) {
        return true;
    }
    
    let minJumps = new Array(nums.length);
    minJumps[0] = 0;
    
    for (let i = 1; i < nums.length; i++) {
        for (let j = 0; j < i; j++) {
            if (i <= j + nums[j]) {
                if (!minJumps[i]) {
                    minJumps[i] = minJumps[j] + 1;
                } else {
                    minJumps[i] = Math.min(minJumps[i], minJumps[j] + 1);
                }
            }
        }
        if (!minJumps[i]) {
            return false;
        }
    }
    
    return true;
};

File: leetcodeJava/src/main/java/Problem55.java
// 55. Jump Game

class Solution {
    public boolean canJump(int[] nums) {
        if (nums.length == 1) {
            return true;
        }
        
        int maxIndex = nums[0];
        for (int i = 0; i < nums.length - 1; i++) {
            if (i > maxIndex) {
                return false;
            }
            maxIndex = Math.max(maxIndex, nums[i] + i);
            if (maxIndex >= nums.length - 1) {
                return true;
            }
        }
        
        return false;
    }
}